### DISCUSSION TITLE
Is It possible  to download offline Polygon shaped tile instead of  rectangular  size ?

### DISCUSSION THREAD

**Discussion by akunal1** [rnmapbox/maps#1295] [Q&A]
Hey,

I have to download thousands of area (small areas representing  fields) from different regions. To minimize downloaded  data I want to download Aras in its shape (polygon, triangular, square, etc...) .

But current `<MapboxGL.offlineManager /> ` is only accepting 4 coordinate points. Do we have any options to accomplish this ? 

Thanks.

**Comment by devWaleed** [comment]
You can use turf.js lineString along with bbox to make a square around your field. Then from that bbox, you can take its NE/SW coordinates and download tiles between those two coords only. 

However since the zoom ranges are grouped, you will still be downloading a small amount of unnecessary tiles from server.

```
const lineStringArray = points?.map?.((p) => {
        return [p.longitude, p.latitude];
      });

      const line = lineString(lineStringArray);

      // Create a square box around line string
      const bounds = bbox(line);
```
